pkg/bosh/converter/job_factory.go

```diff
@@ -302,14 +283,26 @@ func (f *JobFactory) releaseImageQJob(name string, manifest bdm.Manifest, secret
 	}
 
 	outputMap := qjv1a1.OutputMap{}
-	outputSecretNamePrefix := names.DeploymentSecretPrefix(secretType, manifest.Name)
+	igPrefix := names.DeploymentSecretPrefix(names.DeploymentSecretTypeInstanceGroupResolvedProperties, manifest.Name)
+	bpmPrefix := names.DeploymentSecretPrefix(names.DeploymentSecretBpmInformation, manifest.Name)
 	for _, container := range containers {
-		outputMap[container.Name] = qjv1a1.NewFileToSecret(
-			outputFilename,
-			// the same as names.InstaceGroupSecretName(secretType, manifestName, container.Name, "")
-			outputSecretNamePrefix+container.Name,
-			true,
-		)
+		outputMap[container.Name] = qjv1a1.FilesToSecrets{
+			InstanceGroupOutputFilename: qjv1a1.SecretOptions{
+				// the same as names.InstanceGroupSecretName(secretType, manifestName, container.Name, "")
+				Name: igPrefix + container.Name,
+				AdditionalSecretLabels: map[string]string{
+					bdv1.LabelDeploymentSecretType: names.DeploymentSecretTypeInstanceGroupResolvedProperties.String(),
+				},
+				Versioned: true,
+			},
+			BPMOutputFilename: qjv1a1.SecretOptions{
+				Name: bpmPrefix + container.Name,
```

Comment:
Shouldn't this name be constructed in `names`?

Reply:
the base is constructed in `names`, inside the loop the container name is added. I just copied that behavior from the other output. I think it's an optimization, so I'd leave it at that for now.